errata-email-notifications.py

```diff
@@ -0,0 +1,247 @@
+# -*- coding: utf-8 -*-
+import argparse
+import json
+import logging
+import os
+import smtplib
+import ssl
+import sys
+
+from datetime import datetime
+from email.message import EmailMessage
+from string import Template
+from urllib import request
+
+# For simplicity, we store files in the same folder as the script lives in.
+# These files are:
+#  * [distribution]_last_processed_ts: One file per distribution
+#  * app-passwd: GMail application password
+# For this reason, we'll use BASEPATH anytime we read/write files and
+# can be changed in the future if needed
+BASEPATH = os.path.dirname(__file__)
+
+DISTRIBUTIONS_ERRATA_URL = {
+    'almalinux-8': 'https://errata.almalinux.org/8/errata.json',
+    'almalinux-9': 'https://errata.almalinux.org/9/errata.json'
+}
+
+EMAIL_FROM_NAME = 'AlmaLinux Errata Notifications'
+# We create the templates for both subject and content beforehand
+SUBJECT_TEMPLATE = Template('[$errata_type Advisory] $errata_title')
+# TODO: Read content template from file?
+CONTENT_TEMPLATE = Template(
+"""Hi, you are receiving an AlmaLinux $errata_type update email because you subscribed to receive errata notifications from AlmaLinux.
+
+Type: $errata_type
+Severity: $errata_severity
+Release date: $errata_date
+
+Summary:
+
+$errata_description
+
+Full details, updated packages, references, and other related information: $errata_link
+
+This message is automatically generated, please don’t reply. For further questions, please, contact us via the AlmaLinux community chat: https://chat.almalinux.org/.
+Want to change your notification settings? Sign in and manage mailing lists on https://lists.almalinux.org.
+
+Kind regards,
+AlmaLinux team
+""")
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        'errata-email-notifications.py',
+        description='AlmaLinux errata email notifications script. Parses public errata files '
+                    'and sends email notifications to the recipient provided.'
+    )
+    parser.add_argument(
+        '-d', '--distribution', type=str, nargs='+', required=True,
+        help='Distribution(s) to fetch/send errata notifications'
+    )
+    # TODO: Add email verification such as format or even
+    # check that it is a valid email address. py3-validate-email
+    # python package can help here.
+    parser.add_argument(
+        '-s', '--sender', type=str, required=True,
+        help='Email address to use to send the emails'
+    )
+    parser.add_argument(
```

Comment:
Can we have a parameter that would take several recipients' e-mails at the time? File with those or something like that? Because fetching and reading errata data for each send looks quite expensive

Reply:
I don't think we need several several recipients here. This script is for sending emails to maillist.

Reply:
I do agree with you @Korulag, but as @andrewlukoshko mentioned, this script is going to send the emails to announce@lists.almalinux.org only. In any case, doing such thing would not involve big changes in the source code, so it can be implemented anytime. I'll leave some notes in the code just in case. Thanks